### ISSUE TITLE
User added data catalog bug

### ISSUE THREAD

**Issue by soyarsauce** [TerriaJS/terriajs#4468]
I thought this was already raised but can't find it, link/close this ticket if it can be found
Currently you can't traverse the "my data" tab of catalog explorer without it kicking you back to "data catalogue" on every click, if you add remote datasets that already exist in the catalog

repro:
1. http://ci.terria.io/mobx/
2. add web data
3.  add something e.g. http://services.ga.gov.au/gis/rest/services/DEM_SRTM_1Second_Aspect/MapServer
4. try expanding / viewing layers 
5. observe it tabbing to "data catalogue" on each view

**Comment by zoran995** [comment]
#4285

**Comment by zoran995** [comment]
Little digging of this showed that the primary source is in `addedByUser` function (more specifically lines [L22-25](https://github.com/TerriaJS/terriajs/blob/mobx/lib/Core/addedByUser.ts#L22-L25)). But the actual cause of the issue can be hidden deeper in the code. Here is a description of what I discovered.

Most of the datasets will have `sourceReference` defined but all elements of `sourceReference.knownContainerUniqueIds` are undefined, while the actual identification is it userAddedData is in `catalogMember.knownContainerUniqueIds`.

Above is the situation for the catalogItems and first-level groups. When it comes to the items inside the group, I identified that they would mostly have undefined `sourceReference` and the `catalogMember.knownContainerUniqueIds` contains the upper-level id.
